Add changelog

adambutler opened the pull request:
## Description

Adds a changelog with `v1.0.0` document.

## Deploy Notes

None

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+
+Having great tooling is incredibly important for being agile, iterative and innovative. Prior to Nexmo Developer the documentation for Nexmo APIs were at <https://docs.nexmo.com>, a site that has served its purpose well over the last four years.
+
+As we look to evolve our documentation beyond just guides our statically generated documentation site didn't seem like the best option for moving forward with. We needed something that accelerate us, have the power to be dynamic but retain the simplicity that we've come to love from file based static site generators.
```
Change second sentence to:. "We needed something to help us acceleate, with the power to be dynamic without losing the simplicity we have come to love from file-based static site generators."

tommorris replied:
look → looked, to fit with the tense of the rest of the sentence.

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+### A bit of history
+
+Having great tooling is incredibly important for being agile, iterative and innovative. Prior to Nexmo Developer the documentation for Nexmo APIs were at <https://docs.nexmo.com>, a site that has served its purpose well over the last four years.
+
```
were → was

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+## Introducing Nexmo Developer
+
+Nexmo Developer is the new home for our documentation, API references, community content & resources. Our team has been working hard to improve our developer experience whether you're new to Nexmo APIs or a well accustomed with our platform and tooling.
+
```
& → and

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+| ### Simple
+|
+| The NDP should serve as the clearest example of what our products are and how to implement them. Documentation & examples should contain no more than the minimum amount of information the developer needs to understand to implement the product, further information should be signposted and hosted elsewhere within the documentation or externally where appropriate.
+
```
& → and

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+| ### Developer centric
+|
+| The NDP should be built with the developer in mind. The site should be designed to be clear, concise, technical and to the point. Furthermore effort should be focused on what is really important for the developer and the context & environment that the documentation will be consumed. For example inaccuracies or ambiguities should be resolved ahead of superficial changes to the mobile styling.
+
```
& → and

tommorris replied:
Comma after "For example".

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+
+| ### Easy to navigate & Consistent
+|
+| The information architecture of NDP should be designed as such to allow developers to find the most appropriate information as easily as possible.
```
Remove "as such".

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+|
+| The information architecture of NDP should be designed as such to allow developers to find the most appropriate information as easily as possible.
+|
+| By enforcing a clear consistent structure throughout the documentation content will become easier to find and quickly retrievable at a later date.
```
Change "clear consistent" to "clear and consistent". Not sure what "quickly retrievable" means.

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+|
+| By enforcing a clear consistent structure throughout the documentation content will become easier to find and quickly retrievable at a later date.
+|
+| A style guide will be created outlining the structure in more detail, the guide requirements and the tone of voice for the NDP copy.
```
Change "guide requirements" to "requirements for guides".

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+
+| ### Accessible
+|
+| Accessibility is about catering for developers with different spoken languages and programming language preferences but most importantly is to enable developers of a varying range of technical expertise. The NDP information architecture should be driven by this providing documentation appropriate for all developers.
```
Different spoken languages? Should we mention that? That sort of commits us to providing non-English documentation. (Which would be lovely.)

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+
+| ### Shareable
+|
+| We need to ensure that documentation can be referenced on support platforms. You should be able to hover over each heading, subheading or example to create an anchored permalink.
```
"referenced on support platforms": change "on" to "from". I'm presuming support platforms means both email support (Zendesk) and things like StackOverflow. We might want to specify that's what we mean.

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+|
+| We need to ensure that documentation can be referenced on support platforms. You should be able to hover over each heading, subheading or example to create an anchored permalink.
+|
+| This allows us to stay in control of our content by ensuring documentation is always shared by reference as opposed to copying and pasting.
```
Non-novice developers will grasp pass-by-reference vs. pass-by-value but non-devs (or those who haven't studied Russellian linguistic philosophy!) might not know what "by reference" means here.

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+
+| ### Beautiful
+|
+| Beauty is an important tool in conveying the craftsmanship of our product and therefore gaining trust and buy-in. It can also be used as a mechanism for enhancing more in-depth topics to increase engagement.
```
The contrast between the word 'beauty' and the managerial prose needs work. Maybe something like:

> Developers would prefer to read something beautiful. Making documentation and developer resources readable and beautiful increases engagement, trust and makes people feel good about using our product.

adambutler replied:
I don't think "Developers would prefer to read something beautiful" strikes on the point I was trying to make here.

I wanted to convey the that we have considered attention to detail down to the aesthetics and it's effectiveness as communicating the same attention to detail that goes into our products.

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,91 @@
+|
+| Currently documentation exists at <https://docs.nexmo.com>
+|
+| It is important to not break a resource that a developer is consuming & that we maintain support for all URLs that the current docs are using and update the pages with clear notices of where to find the latest documentation that relates to the content of that page.
```
& → and

tommorris replied:
consuming → using

leggetter commented:
Do we still plan to publish this as a blog post and keep the changelog to more factual items?

One thing I think is very important to cover is the documentation types that Nexmo Developer has.

adambutler commented:
> Do we still plan to publish this as a blog post and keep the changelog to more factual items?

Yes I think this should be used as a blog post but I'm also keen to have it in the changelog as well and having more detailed posts like this for all larger releases.

> One thing I think is very important to cover is the documentation types that Nexmo Developer has.

👍 Absolutely agree, I'll add this in.

adambutler commented:
@tommorris Made several tweaks, can you review.

leggetter commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,107 @@
+|
+| A style guide should be maintained outlining the structure in more detail, the requirements for guides and the tone of voice for the Nexmo Developer copy.
+
+| #### Accessible
```
As discussed:

Accessibility is really important and goes beyond semantic machine readable markup and aria roles. Accessibility is also about catering for developers with different programming language preferences and enabling developers of a varying range of technical expertise. The content that we create should be driven by this providing resources appropriate for all ranges of skill level at all stages of a product lifecycle.

leggetter commented on _changelog/1.0.0.md:
> Alongside the guides that we transitioned we created new overview pages for all of our products making it easy to understand their capabilities, to help developers to find the information they needed as easily as possible.

Feels like a list (because of the comma) but we are missing an "and". Can this sentence be reworded?

leggetter commented on _changelog/1.0.0.md:
> We also introduced 'building blocks' as a quick reference to common practical implementations, we hope you'll find these as useful resources during discovery and development phases.

Another comma that, for me, makes this less readable. "which" or "and"?

leggetter commented on _changelog/1.0.0.md:
> it's easier than ever to find the content you need and to keep track of where you are

Remove the "to" in "and to keep track".

leggetter commented on _changelog/1.0.0.md:
> Lastly, we've created new community section

*a* new community section

Link?

leggetter commented on _changelog/1.0.0.md:
> We're excited to grow this to be a hub of community driven content in the future.

Maybe:

We're excited at the potential that this has to be a source of more community information including community driven content.

leggetter commented on _changelog/1.0.0.md:
Unsure how this is making it "more predictable"

leggetter commented on _changelog/1.0.0.md:
I'd move markdown up the chain of importance within the sentence. Is the codebase structure intuitive? Have we created common helpful tags for content to make authoring easier?

leggetter commented on _changelog/1.0.0.md:
IMO "Open Source" deserves its own section covering why we've taken that approach.

leggetter commented on _changelog/1.0.0.md:
"customer Markdown parser" vs. a Markdown parser that intuitively extends "standard Markdown"?

leggetter commented on _changelog/1.0.0.md:
Link to the source.

leggetter commented on _changelog/1.0.0.md:
Search support has been added 🙌 Mention Algolia.

leggetter commented on _changelog/1.0.0.md:
~thorough~

leggetter commented on _changelog/1.0.0.md:
Mention additional benefits of having OpenAPI specs - tooling support that both we and our customers get.

leggetter commented on _changelog/1.0.0.md:
Suggested simplifcation:

As always we welcome feedback. Please let us know using the feedback tool (👍👎) at the bottom of each guide or [file an issue](https://github.com/Nexmo/nexmo-developer/issues/new) on GitHub.

I received feedback (via email 😄 ) that the "Improve this page" and thumbs were a little confusing. I think the addition of the emojis makes it clearer.

adambutler commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,107 @@
+
+### A bit of history
+
+Having great tooling is incredibly important for being agile, iterative and innovative. Prior to Nexmo Developer the documentation for Nexmo APIs was at <https://docs.nexmo.com>, a site that has served its purpose well over the last four years.
```
> Do we want to give a shout out to the platform docs was built on? https://getkirby.com/

I don't think this is necessary and perhaps even detracts. The focus here is on the class of tool as opposed to the specific tool itself.

adambutler commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,107 @@
+
+As we looked to evolve our documentation beyond just guides our statically generated documentation site didn't seem like the best option for moving forward with. We needed something to accelerate us, have the power to be dynamic but retain the simplicity that we've come to love from file-based static site generators.
+
+### Our Goals
```
This is already stated:

> We've made great progress towards making these goals a reality and will continue to iterate and improve on our platform and share our findings along the way.

adambutler commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,107 @@
+
+| #### Developer centric
+|
+| Nexmo Developer should be built with developers in mind. The site should be designed to be clear, concise, technical and to the point. Furthermore effort should be focused on what is *really* important for the developer. Inaccuracies or ambiguities should remain our priority ahead of superficial changes.
```
😂 Fair point

adambutler commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,107 @@
+|
+| The information architecture of Nexmo Developer should be designed to allow developers to find the most appropriate information for their stage of development as easily as possible.
+|
+| By enforcing a clear and consistent structure throughout the documentation content will become easier to find and quickly retrievable at a later date.
```
Good spot!

adambutler commented on _changelog/1.0.0.md:
I've reworded this, let me know what you think.

adambutler commented on _changelog/1.0.0.md:
Reworded slightly, let me know what you think.

adambutler commented on _changelog/1.0.0.md:
It wouldn't be meaningful for it to. Made consistent as per your previous content.

adambutler commented on _changelog/1.0.0.md:
'including' didn't read well as an in place edit. I get your point though and have changed this to:

> The tooling we build should empower other people throughout the business as well as our customers to contribute to Nexmo Developer easily.

--

👍 To your other suggested change

adambutler commented on _changelog/1.0.0.md:
"community information" sounds a bit odd to me.

adambutler commented on _changelog/1.0.0.md:
"more predictable" - I feel as though Nexmo Developer responds to changes in a more predictable and intuitive manner compared to docs making it simpler to contribute to. I think it would be overkill to provide evidence of this and happy for it to remain the simple statement that it is.

adambutler commented on _changelog/1.0.0.md:
Reworded to:

> Nexmo Developer has been built as an open-source platform built using a mix of Ruby on Rails, HTML, JavaScript (ES7 + Webpack), Sass and a document parser that extends Markdown with extra functionality useful for writing technical content.

adambutler commented on _changelog/1.0.0.md:
I don't think this is the place to elaborate on OpenAPI and would prefer to hold this off for a future post.

adambutler commented:
@leggetter Most changes accepted and implemented. A few I've added comments on.

Please can you review ASAP, blog post copy is going to be copy from `1.0.0` with minor tweaks to incorporate changes from `1.1.0`.

tommorris commented on _changelog/1.0.0.md:
* "we've created new community section" → "we've created a new community section"

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,115 @@
+
+## Introducing Nexmo Developer
+
+Nexmo Developer is the new home for our documentation, API references, community content and resources. Our team has been working hard to improve our developer experience whether you're new to Nexmo APIs or a well accustomed with our platform and tooling.
```
> Our team has been working hard to improve our developer experience whether you're new to Nexmo APIs or a well accustomed with our platform and tooling.

This clause feels like it is dangling.

Here's another go at it:

> Our team has been working hard to improve our developer experience: whether you're new to Nexmo APIs or a well accustomed with our platform and tooling, we hope the new Nexmo Developer site makes life easier for you to use our APIs.

adambutler replied:
👍 

Minor tweak, I removed 'we hope'

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,115 @@
+
+Nexmo Developer is the new home for our documentation, API references, community content and resources. Our team has been working hard to improve our developer experience whether you're new to Nexmo APIs or a well accustomed with our platform and tooling.
+
+We will continue to iterate and improve on Nexmo Developer and provide new and innovative ways to help you implement Nexmo APIs into your products quickly, confidently and easily.
```
* Suggestion: "continue to iterate and improve" → "continue to iteratively improve"

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,115 @@
+
+Having great tooling is incredibly important for being agile, iterative and innovative. Prior to Nexmo Developer the documentation for Nexmo APIs was at `docs.nexmo.com`, a site that has served its purpose well over the last four years.
+
+As we looked to evolve our documentation beyond just guides our  generated documentation site didn't seem like the best option for moving forward with. We needed something to accelerate us, have the power to be dynamic but retain the simplicity that we've come to love from file-based static site generators.
```
I don't think most people are going to know what a "generated documentation site" is. Maybe just change it to "existing, mostly statically-generated site"?

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,115 @@
+
+### Our Goals
+
+In January 2017 we started planning out an information architecture that would make content in Nexmo Developer easy and intuitive to find and to provide consistent content across our products.
```
* "In January 2017 we started planning" → "In January 2017, we started planning"
* "to find and to provide" → "to find, and to provide". Oxford comma!

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,115 @@
+
+We arranged distinct content types such as [Documentation](/documentation), [Tutorials](/tutorials) and [API Reference](/api) in our top hand navigation providing quick access directly to the most appropriate content for your stage of development.
+
+We then arranged our documentation to be organised by product (Messaging, Voice, Verify and Number Insight) each of which containing the following sections:
```
* Add a comma: "(Messaging, Voice, Verify and Number Insight) each of" → "(Messaging, Voice, Verify and Number Insight), each of"
* Tense: "containing" → "contain"

tommorris commented on _changelog/1.0.0.md:
```diff
@@ -0,0 +1,115 @@
+
+| #### Developer centric
+|
+| Nexmo Developer should be built with developers in mind. The site should be designed to be clear, concise, technical and to the point. Furthermore effort should be focused on what is *really* important for the developer. Eliminating inaccuracies or ambiguities should remain our priority ahead of superficial changes.
```
* Comma after "Furthermore"
